model_one/cli.py

```diff
@@ -1,198 +1,274 @@
-from array import array
+import os
 import json
-from re import M
-from time import time
 import requests
-import os.path
-import urllib.parse
 
-import os
+from enum import Enum
+from functools import wraps
+from typing import Optional, List, Union
+
+from pandas import Series
+
 
 API_KEY = os.environ.get("BEYONDML_API_KEY")
-api_url = 'https://api.beyond.ml'
+MINIMUM_ENTRIES = 32
 
-def _models_api():
-    return '{}/v0/models'.format(api_url)
 
+class BeyondmlModelException(RuntimeError):
+    pass
 
-def _get_auth_header():
-    return {'Authorization': API_KEY}
 
+class BeyondmlAPI():
+    URL: str = "https://api.beyond.ml"
 
-class BeyondmlModel():
-    def __init__(self, model=None):
-        if model is not None:
-            self._id = model['model_name']
-            self._status = model['status']
-            self._model_type = model['model_type']
-        return
-
-    def __repr__(self):
-        return str({'id': self._id, 'status': self._status, 'model_type': self._model_type})
-
-    def __str__(self):
-        return str({'id': self._id, 'status': self._model_type, 'model_type': self._model_type})
-
-    def _api_post_request(self, handler, data):
-        if not self._id:
-            raise Exception('Initialize the model.')
-        headers = _get_auth_header()
-        headers['Content-Type'] = 'application/json'
-        url = '{}/{}/{}'.format(_models_api(), self._id, handler)
-        r = requests.post(url, headers=headers, data=data)
-        if r.status_code != 200:
-            raise Exception(r.text)
-        self._status = r.json()['status']
-        return r.json()
+    V0: dict = {
+        "create": ("POST", f"{URL}/v0/models"),
+        "models": ("GET", f"{URL}/v0/models"),
+        "generate": ("POST", f"{URL}/v0/{{}}/generate"),
+        "classify": ("POST", f"{URL}/v0/{{}}/classify"),
+        "upload": ("POST", f"{URL}/v0/{{}}/upload"),
+        "status": ("POST", f"{URL}/v0/{{}}/status"),
+        "fit": ("POST", f"{URL}/v0/{{}}/fit"),
+    }
 
-    def fit(self):
-        return self._api_post_request('fit', '')
+    @classmethod
+    def _request(cls, method: str, url: str, params: Optional[dict] = None, data: Optional[dict] = None) -> dict:
```

Comment:
Generic method for:
- Adding api-key to any request
- Raising exception for any non-200 status

Here comes all further shared logic